src/clients/c/tb_client/context.zig

```diff
@@ -560,6 +560,22 @@ pub fn ContextType(
         }
 
         fn on_submit(implementation: *ContextImplementation, packet: *Packet) void {
+            // Packet is caller-allocated to enable elastic intrusive-link-list-based memory
+            // management. However, some of Packet's fields are essentially private. Initialize
+            // them here to avoid threading default fields through FFI boundary.
+            packet.* = .{
+                .next = null,
+                .user_data = packet.user_data,
+                .operation = packet.operation,
+                .status = .ok,
+                .data_size = packet.data_size,
+                .data = packet.data,
+                .batch_next = null,
+                .batch_tail = null,
+                .batch_size = 0,
+                .batch_allowed = false,
```

Comment:
We could simplify this API by moving these fields to a separate struct (opaque to the client libraries).

Example:
```zig
// Exposed to the client.
pub const PacketData = extern struct {
    user_data: ?*anyopaque,
    operation: u8,
    status: Packet.Status,
    data_size: u32,
    data: ?*anyopaque,
    reserved: [42]u8 = [_]u8{0} ** 42, // Internal fields are just opaque.

    comptime {
        assert(@sizeOf(PacketData) == @sizeOf(Packet));
        assert(@alignOf(PacketData) == @alignOf(Packet));
    }
};

// Internal implementation.
pub const Packet = extern struct {
    user_data: ?*anyopaque,
    operation: u8,
    status: Status,
    data_size: u32,
    data: ?*anyopaque,
    next: ?*Packet,
    batch_next: ?*Packet,
    batch_tail: ?*Packet,
    batch_size: u32,
    batch_allowed: bool,
    reserved: [7]u8 = [_]u8{0} ** 7,

    comptime {
        assert(@sizeOf(Packet) == 64);
        assert(@alignOf(Packet) == 8);
    }
}
```